samples/LiveEffect/src/main/cpp/FullDuplexPass.h

```diff
@@ -20,11 +20,14 @@
 
 class FullDuplexPass : public FullDuplexStream {
 public:
-    const int BYTES_PER_FRAME = 4;
     virtual oboe::DataCallbackResult
     onBothStreamsReady(const void *inputData, int numInputFrames, void *outputData,
                        int numOutputFrames) {
-        memcpy(outputData, inputData, numInputFrames * BYTES_PER_FRAME);
+        size_t bytesPerFrame = this->getOutputStream()->getBytesPerFrame();
+        size_t bytesToWrite = numInputFrames * bytesPerFrame;
+        size_t bytesToZero = (numOutputFrames - numInputFrames) * bytesPerFrame > 0 ? : 0;
```

Comment:
An empty case in the ternary operator! Interesting.

 Would bytesToZero be set to 
  ((numOutputFrames - numInputFrames) * bytesPerFrame)
OR
  ((numOutputFrames - numInputFrames) * bytesPerFrame > 0)
I suspect the latter.

An old fashioned if{} might be simpler.